### ISSUE TITLE
Working interactively with Docker

### ISSUE THREAD

**Issue by sstevens2** [dme26/docker-introduction#2]
Hi @dme26!

I added a section to the lessons where I showed them how to interactively work in a docker container.  While it int as reproducible of a way to build images, it is often still useful.  I often work interactively with a container while I'm writing my dockerfile to check that the commands will work as expected.  Someone in the workshop suggested maybe adding a section that mimics that workflow.

It is at the bottom of this lesson, the section called "Creating an image interactively".  It might need to go elsewhere in the lessons.  I also think it might need to be a better example. https://uw-madison-datascience.github.io/docker-introduction/05-creating-container-images/index.html

Let me know what you think.

**Comment by dme26** [comment]
That's a great idea! I've certainly needed to do interactive work to figure out why my Docker containers aren't behaving as I expect. My first thought is maybe to put this as an episode late in the lesson, but can refer to it from early episodes? This way we can have a minimised, "clean" run-through in the early episodes, assuming helpers can ensure everything works for all learners... but we're clear to learners that they might well want to use the later episode if in redoing the lessons privately they get stuck, and given time, the whole group should ideally go through that later episode anyway. The interactive Docker container use has a dependency on *nix shell scripting, although for ResBaz this year, I recall us suggesting everyone should have done the SWC Unix Shell lesson as a prerequisite anyway...

**Comment by sstevens2** [comment]
Indeed.  I did refer to the interactive section a couple of times earlier on when we were building docker files.  Also rushing to prepare last minute, I threw it into the `05-creating-container-images.md` section but I think it should be out on its own and it needs some work to be a bit more of an authentic task.  We could have this as an optional lesson and leave it up to the instructor?
I did require they know the unix shell ahead and suggested they know a little git.  The group I taught it to was bioinformaticians so I expected they might have a bit of experience with the unix shell.  I'm not really sure that ended up being true, also because ~20 extra people snuck in so they didn't see the requirements. 😮 
Also confession I didn't teach the last two lessons...

**Comment by dme26** [comment]
An optional lesson (I think "episode" in Carpentry terms?) sounds ideal.

(Also, that's a nice problem to have in terms of those extra 20 people! :-) )
